tedana/tests/test_integrity _robustica.py

```diff
@@ -0,0 +1,667 @@
+    check_integration_outputs(fn, out_dir, n_logs=2)
+
+
+def test_integration_reclassify_no_bold(skip_integration, caplog):
+    if skip_integration:
+        pytest.skip("Skip reclassify both rejected and accepted")
+
+    test_data_path = guarantee_reclassify_data()
+    out_dir = os.path.abspath(os.path.join(test_data_path, "../outputs/reclassify/no_bold"))
+    if os.path.exists(out_dir):
+        shutil.rmtree(out_dir)
+
+    ioh = InputHarvester(reclassify_raw_registry())
+    comptable = ioh.get_file_contents("ICA metrics tsv")
+    to_accept = [i for i in range(len(comptable))]
+
+    ica_reclassify_workflow(
+        reclassify_raw_registry(),
+        reject=to_accept,
+        out_dir=out_dir,
+        no_reports=True,
+    )
+    assert "No accepted components remaining after manual classification!" in caplog.text
+
+    fn = resource_filename("tedana", "tests/data/reclassify_no_bold.txt")
+    check_integration_outputs(fn, out_dir)
+
+
+def test_integration_reclassify_accrej_files(skip_integration, caplog):
+    if skip_integration:
+        pytest.skip("Skip reclassify both rejected and accepted")
+
+    test_data_path = guarantee_reclassify_data()
+    out_dir = os.path.abspath(os.path.join(test_data_path, "../outputs/reclassify/no_bold"))
+    if os.path.exists(out_dir):
+        shutil.rmtree(out_dir)
+
+    ioh = InputHarvester(reclassify_raw_registry())
+    comptable = ioh.get_file_contents("ICA metrics tsv")
+    to_accept = [i for i in range(len(comptable))]
+
+    ica_reclassify_workflow(
+        reclassify_raw_registry(),
+        reject=to_accept,
+        out_dir=out_dir,
+        no_reports=True,
+    )
+    assert "No accepted components remaining after manual classification!" in caplog.text
+
+    fn = resource_filename("tedana", "tests/data/reclassify_no_bold.txt")
+    check_integration_outputs(fn, out_dir)
+
+
+def test_integration_reclassify_index_failures(skip_integration):
+    if skip_integration:
+        pytest.skip("Skip reclassify index failures")
+
+    test_data_path = guarantee_reclassify_data()
+    out_dir = os.path.abspath(os.path.join(test_data_path, "../outputs/reclassify/index_failures"))
+    if os.path.exists(out_dir):
+        shutil.rmtree(out_dir)
+
+    with pytest.raises(
+        ValueError,
+        match=r"_parse_manual_list expected a list of integers, but the input is",
+    ):
+        ica_reclassify_workflow(
+            reclassify_raw_registry(),
+            accept=[1, 2.5, 3],
+            out_dir=out_dir,
+            no_reports=True,
+        )
+
+    with pytest.raises(
+        ValueError,
+        match=r"_parse_manual_list expected integers or a filename, but the input is",
+    ):
+        ica_reclassify_workflow(
+            reclassify_raw_registry(),
+            accept=[2.5],
+            out_dir=out_dir,
+            no_reports=True,
+        )
+
+
+def test_integration_t2smap(skip_integration):
```

Comment:
This is one of possibly multiple examples where it seems to me that the RobustICA-relevant file includes code that is not specific to RobustICA. It might be that I don't fully see the code path of how the tests are done. But I would expect that in the addition of RobustICA, there would be addition of tests to evaluate exclusively the new RobustICA features, rather than a duplicate "test everything" that includes things not relevant to RobustICA.